modules/db_sqlite/dbase.c

```diff
@@ -824,6 +822,36 @@ int db_sqlite_free_result(db_con_t* _h, db_res_t* _r)
 	return 0;
 }
 
+/**
+ * Release a result set from memory.
+ * \param _r result set whose rows and values should be freed
+ * \return void
+ */
+static void db_sqlite_free_result_rows(db_res_t* _r)
+{
+	db_val_t* values;
+
+	if (!_r) {
+		LM_DBG("nothing to free!\n");
+		return;
+	}
+
+	if(RES_ROWS(_r)!=0)
+	{
+		values = _r->rows[0].values;
+		/* db_sqlite_allocate_rows allocates memory for rows and values separately.
+		 * Hence freeing rows using generic function and then values separately*/
+		db_free_rows(_r);
+		if(values)
+		{
+			pkg_free(values);
+			values = NULL;
+		}
```

Comment:
This entire block can be replaced with just `pkg_free(values);`, for better readability, because:

* `pkg_free()` can handle `NULL` values
* "cleaning up" values by setting is to `NULL` is unnecessary, as it is a short-lived variable (stack only)